### PR TITLE
ofeli: migrate to Conan v2, add v5.1.0

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

File: recipes/ofeli/5.x/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+class OfeliConan(ConanFile):
+    name = "ofeli"
+    description = "An Object Finite Element Library"
+    license = "LGPL-3.0-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://ofeli.org/index.html"
+    topics = ("finite-element", "finite-element-library", "finite-element-analysis", "finite-element-solver")
+
+    package_type = "static-library"
```
Comment (uilianries) [inline recipes/ofeli/5.x/conanfile.py:21]: I did not find a reference offering only static library: https://github.com/rtouzani/ofeli/blob/master/src/CMakeLists.txt

Please, point where you found it.

File: recipes/ofeli/5.x/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD"]:
```
Comment (uilianries) [inline recipes/ofeli/5.x/conanfile.py:38]: The README indicates Windows support too: https://github.com/rtouzani/ofeli/blob/master/README.md

File: recipes/ofeli/5.x/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            raise ConanInvalidConfiguration("Ofeli only supports Linux")
+        if self.settings.compiler != "gcc":
+            raise ConanInvalidConfiguration("Ofeli only supports GCC")
```
Comment (uilianries) [inline recipes/ofeli/5.x/conanfile.py:41]: This is not explicit in the upstream, or I didn't find it documented. Do you have any real reference?

**Review by uilianries — COMMENTED** [review]
Please, check 5.x compatibility. I didn't find those limitations documented. Or they are improved now, or at least, it should be clarified better.

File: recipes/ofeli/5.x/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/ofeli/5.x/conanfile.py:21]: A shared build fails with:
```
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<unsigned long, 3ul>::~LocalVect()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<unsigned long, 2ul>::~LocalVect()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::Muscl::setReconstruction(OFELI::Vect<double> const&, OFELI::Vect<double>&, OFELI::Vect<double>&, unsigned long)'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `virtual thunk to OFELI::LaplaceDG2DP1::build()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 2ul>::get()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 3ul>::LocalVect()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<unsigned long, 3ul>::LocalVect()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<unsigned long, 2ul>::LocalVect()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 3ul>::operator()(unsigned long)'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 2ul>::operator=(double const&)'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 3ul>::get()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 3ul>::~LocalVect()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LaplaceDG2DP1::build()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 1ul>::get()'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 2ul>::operator()(unsigned long)'
ofelia3f8f09b1f9b5/p/lib/libofeli.so: undefined reference to `OFELI::LocalVect<double, 3ul>::operator=(double const&)'
```
on GCC.

File: recipes/ofeli/5.x/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/ofeli/5.x/conanfile.py]: I don't remember the details why I disabled it anymore. Will give it another shot.

File: recipes/ofeli/5.x/conanfile.py (hunk 3 above)
Comment (valgur) [reply recipes/ofeli/5.x/conanfile.py]: Added a comment for it now:
> include/linear_algebra/LocalVect_impl.h:251:42: error: cannot initialize return object of type 'OFELI::Element *' with an lvalue of type 'const OFELI::Element *'

Probably not too difficult to fix, but I would leave it to someone else to take care of. Did not see any PRs for it either.

File: recipes/ofeli/5.x/conanfile.py (hunk 2 above)
Comment (valgur) [reply recipes/ofeli/5.x/conanfile.py]: https://github.com/conan-io/conan-center-index/pull/18952#issuecomment-1794555656
Too many compilation errors on MSVC to count.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**